distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java

```diff
@@ -270,42 +292,42 @@ public int getHitCount(Query query) throws Exception {
         }
     }
 
-    public void processAndPrintResults(DAG[] results, boolean downloadProduct) {
-        LOGGER.info("Printing DAG Attribute Results...");
-        for (int i = 0; i < results.length; i++) {
-            LOGGER.info("\t RESULT : {} of {} ", (i + 1), results.length);
-            printDAGAttributes(results[i]);
-            if (downloadProduct) {
-                try {
-                    retrieveProductFromDAG(results[i]);
-                } catch (MalformedURLException e) {
-                    LOGGER.error("Invalid URL used for product retrieval.", e);
-                }
-            }
+    private void printDagAttributes(DAG[] dags) {
+        for (DAG dag : dags) {
+            printDagAttributes(dag);
         }
     }
 
-    public void printDAGAttributes(DAG dag) {
+    /**
+     * Prints attributes from dag.
+     *
+     * @param dag
+     * @return hashmap of attributes and their values
+     */
+    public Map<String, String> printDagAttributes(DAG dag) {
         LOGGER.info("--------------------");
         LOGGER.info("PRINTING DAG ATTRIBUTES");
-        for (int i = 0; i < dag.nodes.length; i++) {
-            Node node = dag.nodes[i];
+        Map<String, String> attributesMap = new HashMap<String, String>();
+        for (Node node : dag.nodes) {
             if (node.node_type.equals(NodeType.ATTRIBUTE_NODE)) {
                 String name = node.attribute_name;
                 String value = CorbaUtils.getNodeValue(node.value);
                 LOGGER.info("{} = {}", name, value);
+                attributesMap.put(name, value);
             }
         }
         LOGGER.info("--------------------");
+        return attributesMap;
     }
 
-    public void retrieveProductFromDAG(DAG dag) throws MalformedURLException {
+    public void downloadProductFromDAG(DAG dag) throws MalformedURLException {
         LOGGER.info("Downloading products...");
         for (int i = 0; i < dag.nodes.length; i++) {
             Node node = dag.nodes[i];
-            if (node.attribute_name.equals("productUrl")) {
+            if (node.attribute_name.equals(NsiliConstants.PRODUCT_URL)) {
 
                 String url = node.value.extract_string();
+                url = url.replaceAll("\\s+", "%20");
```

Comment:
This may be a bug in the code that sets the URL if it is leaving in spaces vs properly encoding the URL. Also, we should probably be using http://docs.oracle.com/javase/8/docs/api/java/net/URLEncoder.html instead of encoding it ourselves.